src/camera.py

```diff
@@ -0,0 +1,247 @@
+
+        :param degree: The number of degrees.
+        :type degree: int
+        """
+        if self._camera_type == ABSOLUTE:
+            self._current_tilt_angle = np.clip(degree, self.MIN_DEGREE,
+                                                       self.MAX_TILT_DEGREE)
+        else:
+            self._current_tilt_angle = np.clip(degree + self._current_tilt_angle,
+                                                        self.MIN_DEGREE,
+                                                        self.MAX_TILT_DEGREE)
+
+        self._tilt_servo.write(np.clip(self._current_tilt_angle +
+                                       self._tilt_offset,
+                                       self.MIN_DEGREE,
+                                       self.MAX_TILT_DEGREE))
+
+
+    def calibrate_tilt(self, degree):
+        """
+        Calibrates the camera's tilting servo.
+
+        Writes a new tilt offset value to the config file and changes the current
+        tilt offset. The calibration value can be negative.
+
+        :param degree: The number of degrees.
+        :type degree: int
+        """
+        if self._camera_type == ABSOLUTE:
+            self._tilt_offset = int(np.clip(degree, self.MIN_TILT_CALI_DEGREE,
+                                                    self.MAX_TILT_CALI_DEGREE))
+        else:
+            self._tilt_offset = int(np.clip(degree + self._tilt_offset,
+                                                     self.MIN_TILT_CALI_DEGREE,
+                                                     self.MAX_TILT_CALI_DEGREE))
+
+        self._db.set('tilt_offset', self._tilt_offset)
+
+
+    def get_tilt_angle(self):
+        """
+        Gets the current tilt angle of the camera.
+
+        :return: The current tilt angle (in degrees).
+        :rtype: int
+        """
+        return self._current_tilt_angle
+
+
+def main():
+    """
+    Demonstrates the camera's panning and tilting functionality.
+
+    Creates a camera object and pans from 0 to 180, then from 180 to 0. The
+    camera then tilts from 0 to 150, then 150 to 0.
+    """
+    camera_smoothing = 0.05
+    picar.setup()
+    camera = Camera()
+
+    # Tests camera pan.
+    print "Panning left."
+    for i in range(0, 181, 2):
+        camera.pan(i)
+        time.sleep(camera_smoothing)
+
+    print "Panning right."
+    for i in range(180, -1, -2):
+        camera.pan(i)
+        time.sleep(camera_smoothing)
+
+    # Resetting Pan.
+    camera.pan(90)
+    time.sleep(camera_smoothing)
+
+    # Tests camera tilt.
+    print "Tilting up."
+    for i in range(0, 151, 2):
+        camera.tilt(i)
+        time.sleep(camera_smoothing)
+
+    print "Tilting down."
+    for i in range(150, -1, -2):
+        camera.tilt(i)
+        time.sleep(camera_smoothing)
+
+    # Resetting camera for relative pan.
+    cam.pan(0)
+    time.sleep(camera_smoothing)
+
+    # Remaking object in relative style.
+    camera = None
```

Comment:
No need to set the camera to None. Garbage collection takes care of that when you assign it to a new object.